src/App.vue

```diff
@@ -16,12 +16,16 @@ export default defineComponent({
   components: {
     Loading,
   },
-  setup() {
-    const globalStore = useGlobalStore();
-    const loading = computed(() => globalStore.loading);
+  // setup() {
+  //   const globalStore = useGlobalStore();
+  //   const loading1 = computed(() => globalStore.loading);
 
+  //   provide('loading2', loading1)
+  // },
+  provide() {
+    const globalStore = useGlobalStore();
     return {
-      loading,
+      loading: () => computed(() => globalStore.loading),
```

Comment:
because `() => computed(() => globalStore.loading)` is a function, so when evaluate in condition, its always TRUE